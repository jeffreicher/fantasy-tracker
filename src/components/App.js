import React, { Component } from 'react';
import Navbar from './navbar';
import Score from '../container/scoreContainer';

class App extends Component {
    render() {
        return (
            <div className="container-fluid">
                <Navbar />
                <Score />
            </div>
        );
    };
};

export default App;