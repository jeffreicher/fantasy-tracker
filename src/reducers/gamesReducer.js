import { FETCH_GAMES } from '../actions'
export default function(state = [], action) {
  // console.log(action.type)
  switch (action.type) {
    case FETCH_GAMES:
      return {
        ...state,
        games: action.payload,
        noGames: false
      }
    default:
      return state
  }
}
