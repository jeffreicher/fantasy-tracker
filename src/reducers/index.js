import { combineReducers } from 'redux';

const rootReducer = combineReducers({
    // games: GamesReducer
});

export default rootReducer;