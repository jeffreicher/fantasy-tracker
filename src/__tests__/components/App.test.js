import React from 'react'
import ReactDOM from 'react-dom'
import App from '../../components/App'
import { Provider } from 'react-redux'
import reducers from '../../reducers'
import ReduxPromise from 'redux-promise'
import { createStore, applyMiddleware } from 'redux'
import renderer from 'react-test-renderer'

const createStoreWithMiddleware = applyMiddleware(ReduxPromise)(createStore)
describe('tests App component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <Provider store={createStoreWithMiddleware(reducers)}>
        <App />
      </Provider>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })

  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={createStoreWithMiddleware(reducers)}>
          <App />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
