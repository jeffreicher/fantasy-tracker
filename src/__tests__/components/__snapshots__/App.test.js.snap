// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests App component matches snapshot 1`] = `
<div
  className="container-fluid"
>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light justify-content-between"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      Fantasy Tracker
    </a>
    <div
      className="navbar-nav"
    >
      <a
        className="nav-item nav-link"
        href=""
      >
        Scoreboard
      </a>
      <a
        className="nav-item nav-link"
        href=""
      >
        Standings
      </a>
    </div>
  </nav>
  <div>
    NO GAMES FOR TODAY.
  </div>
</div>
`;
